src/main/java/fibonacci/QueueReader.java

```diff
@@ -0,0 +1,39 @@
+package fibonacci;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Created by Nigel on 6/28/16.
+ */
+public class QueueReader implements Runnable {
+    ExecutorService service = Executors.newFixedThreadPool(100);
+    BlockingQueue<Integer> queue = new ArrayBlockingQueue<>(1000);
```

Comment:
You create a new one of these instead of using the one you declared in `main`.  This class should take the executor from in the constructor so that the main thread can watch and control the executor from outside of the thread.